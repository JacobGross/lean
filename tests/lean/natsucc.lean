import data.nat
open nat

theorem add_assoc (m n k : nat) : m + n + k = m + (n + k) :=
nat.induction_on k
(show m + n + 0 = m + (n + 0), from rfl),
(take k,
   assume IH : m + n + k = m + (n + k),
   show m + n + mysucc k = m + (n + mysucc k), from
     calc
       m + n + mysucc k = mysucc (m + n + k) : rfl
       ... = mysucc (m + (n + k)) : IH
       ... = m + mysucc (n + k) : rfl
       ... = m + (n + mysucc k) : rfl)
