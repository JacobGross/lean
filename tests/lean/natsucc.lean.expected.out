natsucc.lean:6:40: error: unexpected token, '.', command, Lua script, or end-of-file expected
